base/src/main/java/com/smartdevicelink/managers/screen/SoftButtonObject.java

```diff
@@ -309,4 +309,20 @@ public void setOnEventListener(OnEventListener onEventListener) {
          */
         void onUpdate();
     }
+
+    /**
+     * @param o
+     * @return true if two SoftButtonObjects have the same name.
+     */
+    @Override
+    public boolean equals(Object o) {
+        if(o == null){
+            return false;
+        }
+        SoftButtonObject softButtonObject = (SoftButtonObject) o;
```

Comment:
This could lead to a class cast exception. We should check `instance of` before casting.

Reply:
I have added fix for this.